### PR TITLE
Model kubernetes SecretInterface

### PR DESCRIPTION
**Pull request by owen-mc**
This can be cleaned up in future, perhaps with a concept of something that is a source of secrets.

### CODE REVIEW

File: ql/src/semmle/go/frameworks/K8sIoClientGo.qll [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Provides models of commonly used functions in the `k8s.io/client-go/kubernetes/typed/core/v1`
+ * package.
+ */
+module K8sIoClientGo {
+  /** Gets the package name. */
+  bindingset[result]
+  string packagePath() { result = package("k8s.io/client-go", "kubernetes/typed/core/v1") }
+
+  /**
+   * A model of go-restful's `Request` object as a source of user-controlled data.
```
Comment (smowton) [inline ql/src/semmle/go/frameworks/K8sIoClientGo.qll:15]: Comment copied from elsewhere

**Review by smowton — COMMENTED** [review]
Looks good, pending tests for the new code It think?

File: ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/go.mod [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module codeql-go-tests/frameworks/K8sIoClientGo
+
+go 1.14
+
+require (
+	github.com/github/depstubber v0.0.0-20201022140002-ee3b8f2acc53 // indirect
```
Comment (sauyon) [inline ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/go.mod:6]: Well, at least `@gagliardetto` has a PR up that should solve this.

File: ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		a, b := s.Watch(ctx, opts)
+		useTwoArgs(a, b)
+	}
+	{
+		a, b := s.Patch(ctx, name, pt, data, opts) // $KsIoClientGo
+		useTwoArgs(a, b)
+	}
+}
+
+func useOneArg(arg interface{})                     {}
```
Comment (sauyon) [inline ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go:56]: You can write `func use(arg interface{}...)` to have a `use` function that takes any number of arguments (though it's unnecessary to change it for this PR).

File: ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+func useTwoArgs(arg1 interface{}, arg2 interface{}) {}
+
+// type mySecretInterface struct {
+// }
+
+// // func (m mySecretInterface) EachListItem(func(runtime.Object) error) error { return nil }
```
Comment (sauyon) [inline ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go:62]: Why are these here?

**Review by sauyon — APPROVED** [review]
A few very minor comments.

File: ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go (hunk 4 above)
Comment (owen-mc) [reply ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go]: Oops, I meant to uncomment them and use `mySecretInterface` in the tests. I've done it now.

**Comment by owen-mc** [comment]
The dist-compare run was fine, so I will set this to auto-merge when the checks pass

File: ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go (hunk 4 above)
Comment (gagliardetto) [reply ql/test/library-tests/semmle/go/frameworks/K8sIoClientGo/main.go]: I have another PR in depstubber that will automatically detect and stub imports, in case you haven't done that manually yet.
